core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -241,14 +252,42 @@ public void testSliceRle()
         }
     }
 
+    @Test
+    public void testSliceAppenderInputNotByteArray()
+    {
+        testAppend(VARCHAR, ImmutableList.of(
+                input(new VariableWidthBlock(3, Slices.wrappedLongArray(257, 2), new int[] {0, 1, Long.BYTES, 2 * Long.BYTES}, Optional.empty()), 0, 2)));
+    }
+
+    @Test
+    public void testSliceAppenderInputRleNotByteArray()
```

Comment:
Isn't this already covered by adaptation?

Reply:
tests with block adaptation

Reply:
Ah, this new one. So no, it's not covered by normal tests, even with adaptation. Like I mentioned [here](https://github.com/trinodb/trino/pull/12798#discussion_r959540741) I need to access `SlicePositionsAppender` directly because `UnnestingPositionsAppender` copies RLE value to byte array always.
That said, in the current implementation I don't rely on the byte array in the RLE block anymore so I could drop this test